api/bundle/client.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package bundle provides access to the bundle api facade.
+// This facade contains api calls that are specific to bundles.
+
+package bundle
+
+import (
+	"github.com/juju/loggo"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/apiserver/params"
+)
+
+var logger = loggo.GetLogger("juju.api.application")
+
+// Client allows access to the application API end point.
+type Client struct {
+	base.ClientFacade
+	st     base.APICallCloser
+	facade base.FacadeCaller
+}
+
+// NewClient creates a new client for accessing the bundle api.
+func NewClient(st base.APICallCloser) *Client {
+	frontend, backend := base.NewClientFacade(st, "Bundle")
+	return &Client{
+		ClientFacade: frontend,
+		st:           st,
+		facade:       backend}
+}
+
+// ExportBundle exports the current model configuration.
+func (c *Client) ExportBundle(tag names.ModelTag) (params.StringResult, error) {
```

Comment:
Methods on the api wrappers should unpack the params structs. Ideally there would never be a params struct returned from an API method.

So in this case return `(string, error)`, and get the string value out of the StringResult struct.

Reply:
Also... the concept to Tags was supposed to be primarily for the wire protocol for calling methods, and for accessing objects in the persistence layer.

There is no need to pass in a model tag because the API is limited to a model itself. There is no argument to pass. The ExportBundle method on the facade takes no args.

Reply:
You also want an early exit if the best facade version < 2 with a nice error.